File: 2015/1-not-quite-lisp/day1p2.py
filename = './input.txt'

with open(filename) as f:
    source = [line.rstrip('\n') for line in open(filename)]

floor = 0
count = 0

for line in source:
    for char in line:
        if char == '(':
            floor = floor + 1
        else:
            floor = floor - 1
        count = count + 1
        if floor == -1:
            print "Santa enters the basement on instruction", count
            break

File: 2015/1-not-quite-lisp/day1p2.js
var fs = require('fs')

var source = fs.readFileSync('../source/day1source.txt').toString()
var end = source.length
var floor = 1
for (var i = 0; i < end; i++) {
  if (source.charAt(i) === '(') {
    floor++
  } else {
    floor--
  }
  if (floor === -1) {
    console.log('The character ' + source.charAt(i) + ' at position ' + i + ' sent santa to the basement.')
  }
}
